lib/src/shared/main/java/com/couchbase/lite/Blob.java

```diff
@@ -283,36 +295,52 @@ public Blob(@NonNull String contentType, @NonNull URL fileURL) throws IOExceptio
     //---------------------------------------------
 
     /**
-     * Gets the contents of a Blob as a block of memory. Not recommended for very large blobs, as it
-     * may be slow and use up lots of RAM.
-     * <p>
+     * Gets the contents of a Blob as a block of memory.
+     * Not recommended for very large blobs, as it may be slow and use up lots of RAM.
      *
      * @return the contents of a Blob as a block of memory
      */
-     // !!! FIXME: This method returns a writable copy of its private data
-    @SuppressFBWarnings("EI_EXPOSE_REP")
+    @Nullable
     public byte[] getContent() {
-        if (content != null) {
-            // Data is in memory:
-            return content;
-        }
+        if (content != null) { return copyBytes(content); }
+
+        if (database != null) { return getBytesFromDatabase(); }
 
-        return (database != null) ? getBytesFromDatabase() : getBytesFromInitialContentStream();
+        // data must be in the initial content stream
+        // No recourse but to read it
+        if (initialContentStream == null) { throw new IllegalStateException("Initial content stream is null"); }
+        try { return getBytesFromStream(initialContentStream); }
+        finally {
+            try { initialContentStream.close(); }
+            catch (IOException e) { Log.i(DOMAIN, "Failed reading blob content stream", e); }
+            initialContentStream = null;
```

Comment:
This means that calling getContent() second time could return null if the content > MAX_CACHE_SIZE. 

Potential issue:
```
Blob b = new Blob(type, is); // blob is larger than 8M
byte[] c = b.content; // this will cause the initial stream to be null but its content is not cached.

Document doc = new Document();
d.setBlob(key, b);
database.save(doc); // when install blob into the database, I think the content might be empty
```

Reply:
Yes.  This is what it has always done.  That is CBL-390.  What should it do in this case?  Re-read the stream?

Reply:
So... this is not a regression.  What would you think of approving this PR with the understanding that it does not fix the bug?  This is just code cleanup.  Bug fix is next.

Reply:
Yes